edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubeUtils.cs

```diff
@@ -86,13 +86,19 @@ public static string SanitizeK8sValue(string name)
             return output.ToString();
         }
 
-        // DNS label (as per RFC 1035)
-        public static string SanitizeDNSValue(string name)
+        /// DNS label (as per RFC 1035
+        public static string SanitizeDNSValue(string name) => SanitizeDNSValue(name, IsAlpha);
+
+        // DNS label (as per RFC 1035/1123)
+        public static string SanitizeDNSLabel(string name) => SanitizeDNSValue(name, IsAlphaNumeric);
+
+        private static string SanitizeDNSValue(string name, Func<char, bool> startCriteria)
```

Comment:
Is system.uri.checkhostname sufficient? https://docs.microsoft.com/en-us/dotnet/api/system.uri.checkhostname?redirectedfrom=MSDN&view=net-5.0

Reply:
Back when we wrote this, we debated on using these tools, but we had distinct requirements, specifically, we needed to drop the "$" from the system modules, "$edgeHub" -> edgeHub, so sanitizing rather than checking made more sense.

Reply:
Okay. Thank you!